cocotb/handle.py

```diff
@@ -836,6 +849,11 @@ def _set_value(self, value, call_sim):
     def value(self) -> bytes:
         return self._handle.get_signal_val_str()
 
+    def __str__(self):
+        # Skip the deprecated overload in ModifiableObject, which is nonsense
+        # for bytes.
+        return SimHandleBase.__str__(self)
```

Comment:
```suggestion
        # default to ASCII representation as is done when setting values
        return self.value.decode('ascii')
```

Reply:
This should have the same deprecation warning as its parent class then.

Reply:
Atually, this is made redundant by your implementation above, right?

Reply:
Ah I see, this was needed after all.